lib/shared/bucketing-assembly-script/assembly/eventQueue/requestPayloadManager.ts

```diff
@@ -101,32 +101,59 @@ export class RequestPayloadManager {
             }
         }
 
-        // Generate defaulted agreagate user as our events APIs require a user / user_id
+        // Generate defaulted aggregate user as our events APIs require a user / user_id
         const dvcUser = new DVCPopulatedUser(new DVCUser(
             user_id, null, null, null, null, NaN, null, null, null, null
         ))
         this.addEventsToPendingPayloads(new UserEventsBatchRecord(dvcUser, aggEvents))
     }
 
+    /**
+     * Find existing FlushPayload that contains records for the user_id,
+     * or return the smallest FlushPayload or a new FlushPayload
+     */
+    private getFlushPayloadForUser_id(user_id: string): FlushPayload {
+        const keys = this.pendingPayloads.keys()
+        let smallestPayload: FlushPayload | null = null
+
+        for (let i=0; i < keys.length; i++) {
+            const key = keys[i]
+            const payload = this.pendingPayloads.get(key)
+            const recordForUser = payload.recordForUser(user_id)
+            const payloadEventCount = payload.eventCount()
+            if (recordForUser && payloadEventCount < this.chunkSize) {
+                this.pendingPayloads.delete(key)
```

Comment:
why does the found payload need to be deleted from pending payloads before being returned? Is it because you're then going to add more stuff to it? Maybe another good reason they should be immutable?

Reply:
yea basically because the calling code always adds it. I'll try and restructure this to remove this delete.